crates/core/src/doc/table.rs

```diff
@@ -343,38 +963,143 @@ impl Document {
 		data.cut(ID.as_ref());
 		Ok(Data::ReplaceExpression(data.into_literal()))
 	}
-	//
-	async fn data(
+
+	#[allow(clippy::too_many_arguments)]
+	async fn handle_record_with_metadata(
 		&self,
 		stk: &mut Stk,
 		ctx: &Context,
 		opt: &Options,
-		fdc: FieldDataContext<'_>,
+		rid: &RecordId,
+		set_ops: Vec<Assignment>,
+		del_ops: Option<Expr>,
+		metadata_deltas: HashMap<String, FieldStatsDelta>,
 	) -> Result<()> {
-		//
-		let (set_ops, del_ops) = self.fields(stk, ctx, opt, &fdc).await?;
-		//
-		let thg = RecordId {
-			table: fdc.ft.name.clone(),
-			key: RecordIdKey::Array(Array(fdc.group_ids)),
-		};
-		let what = vec![Expr::Literal(Literal::RecordId(thg.clone().into_literal()))];
-		let stm = UpsertStatement {
-			what,
-			data: Some(Data::SetExpression(set_ops)),
-			..UpsertStatement::default()
+		use crate::expr::FlowResultExt as _;
+
+		// Get NS & DB identifiers
+		let (ns, db) = ctx.expect_ns_db_ids(opt).await?;
+
+		// Get the transaction
+		let txn = ctx.tx();
+
+		// Clear cache to ensure we get the latest version
+		txn.clear_cache();
```

Comment:
Is this required? If the cache does not reflect the current state of the KV store is that not a bug?